source/Ship.cpp

```diff
@@ -1980,8 +1982,8 @@ void Ship::WasCaptured(const shared_ptr<Ship> &capturer)
 	// Transfer some crew over. Only transfer the bare minimum unless even that
 	// is not possible, in which case, share evenly.
 	int totalRequired = capturer->RequiredCrew() + RequiredCrew();
-	int transfer = RequiredCrew();
-	if(transfer)
+	int transfer = RequiredCrew() - crew;
+	if(transfer > 0)
 	{
 		if(totalRequired > capturer->Crew())
 			transfer = max(1, (capturer->Crew() * RequiredCrew()) / totalRequired);
```

Comment:
If you're adding support for capturing a ship that still has crew, I think this would need to be:

```C++
if(totalRequired > capturer->Crew() + crew)
    transfer = max(crew ? 0 : 1, (capturer->Crew() * transfer) / totalRequired);
```

That is, if there's not enough crew to go around, split the crew from both ships evenly. And, if the victim has no crew, make sure to give it at least one.

Reply:
You are correct